src/Domain/DonationRepository.php

```diff
@@ -670,4 +671,20 @@ private function persistQueuedDonations(): void
             $this->getEntityManager()->persist($donation);
         }
     }
+
+    /**
+     * Locks row in DB to prevent concurrent updates
+     */
+    public function findAndLockOneBy(array $criteria, ?array $orderBy = null): ?Donation
+    {
+        $donation = $this->findOneBy($criteria, $orderBy);
```

Comment:
@bdsl I think it would be good here just to add a code comment pointing out the reason to first select without a lock, for quick reference.

Reply:
Done.